canvas_data_warehouse/views.py

```diff
@@ -0,0 +1,21 @@
+'''Provide API views for data-graph module.'''
+from rest_framework.response import Response
+from rest_framework.decorators import action, list_route
+from rest_framework import status, viewsets
+
+from canvas_data_warehouse.services import CanvasDataService
+
+
+class CanvasDataViewSet(viewsets.ViewSet):
+    '''create API views for getting graph data'''
+    @list_route(name='get-canvas-data')
+    def get_canvas_data(self, request):
+        '''getting data-graph data'''
+        CanvasDataService.dispatch(request.GET.get('graph_type'), request.GET)
```

Comment:
Check whether `graph_type` exists, if not, raise `BadRequest`.
Do not pass `request.GET` directly, convert it to a `dict`, although they may similar interfaces.